### PR TITLE
Integrate with Textile Hub Lambda

### PR DESCRIPTION
**Pull request by dmerrill6**
# Change log

- Replaces authentication to Textile hub so that keys aren't stored in the client side.

- Connects to a lambda through web sockets to get the auth token.

[ch16547]

### CODE REVIEW

File: config/map_config.go [hunk 1]
```diff
@@ -19,7 +19,8 @@ func NewMap(envVal env.SpaceEnv, flags *Flags) Config {
 	configStr[SpaceStorePath] = "~/.fleek-space"
 	configStr[TextileHubTarget] = "textile-hub-dev.fleek.co:3006"
 	configStr[TextileThreadsTarget] = "textile-hub-dev.fleek.co:3006"
-	configStr[SpaceServicesAPIURL] = "https://td4uiovozc.execute-api.us-west-2.amazonaws.com/dev" // TODO: Get a domain
+	configStr[SpaceServicesAPIURL] = "https://td4uiovozc.execute-api.us-west-2.amazonaws.com/dev"   // TODO: Get a domain
```
Comment (perfectmak) [inline config/map_config.go:22]: we should probably pass this values through the build command/flags like the mongo information shown below.
So it is easily configurable on build

File: core/textile/client.go [hunk 2]
```diff
@@ -102,41 +100,18 @@ func (tc *textileClient) requiresRunning() error {
...
-	}
-
-	// TODO: CTX has to be made from session key received from lambda
-	log.Debug("Creating libp2p identity")
-	tok, err := tc.threads.GetToken(ctx, thread.NewLibp2pIdentity(privateKey))
+	log.Debug("Authenticating with Textile Hub")
+	tokStr, err := getHubToken(tc.store, tc.cfg)
 	if err != nil {
-		return nil, err
+		log.Error("Token Challenge Error:", err)
```
Comment (perfectmak) [inline core/textile/client.go:112]: Shouldn't you return the err here, or is it possible to initialize `thread.Token()` with an empty `tokStr`?

File: config/map_config.go (hunk 1 above)
Comment (jsonsivar) [reply config/map_config.go:22]: Yeah this sounds good, maybe for the Textile Hub targets as well.

File: core/textile/client.go (hunk 2 above)
Comment (jsonsivar) [reply core/textile/client.go:112]: Agreed

Comment (dmerrill6) [reply core/textile/client.go:112]: Fixed in 4ef068e

File: config/map_config.go (hunk 1 above)
Comment (dmerrill6) [reply config/map_config.go]: Fixed in 4ef068e

**Pull request merged**